plugin/client.go

```diff
@@ -13,12 +13,10 @@ import (
 func ClientMain(pluginImplementation interface{}) {
 	if impl, ok := pluginImplementation.(interface {
 		SetAPI(api API)
```

Comment:
It's interesting to observe that the only reason to embed `MattermostPlugin` is for `SetAPI`. In theory, we could make the panic here optional and give plugins even more flexibility.